src/lib/frontend/d_cnf.ml

```diff
@@ -0,0 +1,1143 @@
+
+open Options
+open D_loop
+
+module E = Expr
+module SE = E.Set
+
+module C = Commands
+module Sy = Symbols
+module SM = Sy.Map
+
+module DE = DStd.Expr
+module DT = DE.Ty
+module Id = DStd.Id
+module B = DStd.Builtin
+
+module ME =
+  Map.Make
+    (struct
+      type t = E.t
+      let compare a b =
+        let c = E.depth a - E.depth b in
+        if c <> 0 then c
+        else E.compare a b
+    end)
+
+module HT = Hashtbl.Make(
+  struct
+    type t = int
+    let equal = (=)
+    let hash = Fun.id
+  end)
+
+module TSM = Map.Make (
+  struct
+    type t = int
+    let compare = Int.compare
+  end
+  )
+
+(** Helper function: returns the basename of a dolmen path, since in AE
+    the problems are contained in one-file (for now at least), the path is
+    irrelevant and only the basename matters *)
+let get_basename DStd.Path.(Local { name; } | Absolute { name; _ }) =
```

Comment:
you should at the very least, assert that the absolute path is empty, or else that will cause problems later

Reply:
Done.